### PR TITLE
Update to beta.17

### PR DESCRIPTION
**Pull request by cavis**
Mostly fixing deprecated stuff, and more major changes in how Router and ViewChild work.

Also got rid of the `jspm:unbundle` pre-commit hook.  Development mode now just creates a 2nd (ignored) `config/systemjs.dev.js` file, which contains just the vendor bundle.

### CODE REVIEW

File: app/footer/footer.component.ts [hunk 1]
```diff
@@ -52,12 +51,12 @@ import {Location} from 'angular2/router';
 
 export class FooterComponent {
 
-  constructor(
-    private location: Location
-  ) {}
+  locationPath(): string {
```
Comment (farski) [inline app/footer/footer.component.ts]: this could be a `get`

Comment (cavis) [inline app/footer/footer.component.ts]: For some reason, jasmine's `spyOn` doesn't work on getters.  So if I want to mock it, it has to be an actual function.

This was more straight forward before they took `Location` out of the router.

Comment (farski) [inline app/footer/footer.component.ts]: huh. bummer

**Comment by farski** [comment]
Seems okay to me, will let @kookster review and merge

**Comment by kookster** [comment]
lgtn 👍

**Pull request merged**